### ISSUE TITLE
Importing enpass json - Error: invalid credentials, password encryption/decryption aborted.

### ISSUE THREAD

**Issue by codecleric** [roddhjav/pass-import#114]
I'm confused on this one - the file exported fine, and my pass store is working as expected. Any thoughts?

`pass import -c -a -v enpass enpassout.json`

results in:
` [x] Error: invalid credentials, password encryption/decryption aborted.`

**Comment by roddhjav** [comment]
Hi, Can you run it with the `-vvv` option instead of `-v` and provide the output. (Beware, this will also print the data you import, so do not forget to remove them)

**Comment by codecleric** [comment]
I'll double check the keys and the trust. Here is the output of -vvv:
```
  .  Trying to guess file format.
  .  Importer: enpass, Format: json, Version: 6
  .  Importing passwords from Enpass6 to PasswordStore
  .  Traceback (most recent call last):
  File "/home/brian/.local/lib/python3.8/site-packages/pass_import-3.0-py3.8.egg/pass_import/__main__.py", line 358, in pass_export
    with cls_export(conf['out'], settings=settings) as exporter:
  File "/home/brian/.local/lib/python3.8/site-packages/pass_import-3.0-py3.8.egg/pass_import/core.py", line 116, in __enter__
    raise PMError(
pass_import.errors.PMError: invalid credentials, password encryption/decryption aborted.

 [x] Error: invalid credentials, password encryption/decryption aborted.
```

**Comment by codecleric** [comment]
All I had to do was trust one the keys in my keyring and everything is working fine now. Thanks for the hint!

**Comment by sluedtke** [comment]
Hi there, 
sorry for reopening that one again I get the same errror message (from firefox and csv import) but I do not understand what @codecleric suggested to solve the problem.

**Comment by roddhjav** [comment]
Hi, I will add a section to the readme but, the issue is not related to pass-import or even to pass. It is your GPG keyring that is not configured. Before importing password in your password repository, pass-import ensures your gpg keyring is usable for this repository. You need to ensure that:
- All the public gpgids are present in the keyring.
- All the public gpgids are trusted enough.
- At least one private key is present in the keyring.

So the issue is usually because one of your gpg key is not trusted enough. To do this run `gpg --edit-key <gpgid>` then `trust`.

**Comment by sluedtke** [comment]
Hi, 
as expected, you are right. It was an issue with my key config. I think it does not make sense to go into details, but the import is working fine now. 
thanks

**Comment by allista** [comment]
Sorry, but this still doesn't make sense for me: I have multiple public keys, but only one of them is used in the pass store, and it is trusted and is the default key. Do I have to change trust level for all _unrelated_ keys as well?

**Comment by roddhjav** [comment]
You need to trust all the keys in `.gpgid`. Other keys are not concerned.

**Comment by allista** [comment]
I have temporarily _deleted_ all other public keys, only my own key pair is left.
I don't know what else to do:

```
> gpg --list-public-keys 
-------------------------------
pub   rsa3072 2021-08-03 [SC]
      FDB025C059626752859FFB3E03A8DF3F5E421E83
uid           [ultimate] Evgeny Alexandrovich Taranov <evgeny.taranov@gmail.com>
sub   rsa3072 2021-08-03 [E]
sub   rsa3072 2021-08-03 [A]

> cat .gpg-id 
0x03a8df3f5e421e83

> pimport -vvv -a -c -C pass ./keepassx.kdbx 
  .  Trying to guess file format and manager name.
  .  Importer: keepass, Format: kdbx, Version: 
  .  Importing passwords from Keepass to PasswordStore
Password for keepassx.kdbx: 
  .  Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pass_import/__main__.py", line 371, in pass_export
    with cls_export(conf['out'], settings=settings) as exporter:
  File "/usr/local/lib/python3.8/dist-packages/pass_import/core.py", line 115, in __enter__
    raise PMError(
pass_import.errors.PMError: invalid credentials, password encryption/decryption aborted.

 [x] Error: invalid credentials, password encryption/decryption aborted.
```

**Comment by allista** [comment]
@roddhjav I have found the problem

In `/pass_import/managers/passwordstore.py:PasswordStore.isvalid()`

```
# L:222
        with open(os.path.join(self.prefix, '.gpg-id'), 'r') as file:
            gpgids = file.read().split('\n')
            gpgids.pop() # << this here removes the only key ID there is
```

**Comment by roddhjav** [comment]
Can you run: `gpg --with-colons --batch --list-keys <gpgid>`
In the colons result, you should get something that has `pub:u`, meaning that you have the public key and you trust it ultimately.
Also, when you run `gpg --with-colons --batch --list-secret-keys <gpgid>` you should not raise an error.

**Comment by roddhjav** [comment]
> I have found the problem

Ok, that is different, `pass init` write/read the `.gpg-id` with a new line at the end. It is expected to be there. So you might have edited the file manually.

**Comment by allista** [comment]
I have not: either qtpass or the AndroidPasswordStore did.
And of course inserting it did help.

I would suggest a simple fix for this:
```
        trusted = ['m', 'f', 'u', 'w', 's']
        with open(os.path.join(self.prefix, '.gpg-id'), 'r') as file:
            # first strip file content from white spaces, then split it
            gpgids = file.read().strip().split('\n')
```

I'd make a PR, but have only mobile with me for the next two weeks.

**Comment by roddhjav** [comment]
You are right, I will do it.

**Comment by fk0** [comment]
I believe after a year we still don't have a PR. @allista can you please make one?

**Comment by fk0** [comment]
Just for those who came here with the same error: I've got this issue because I screwed the initial setup, and instead if doing 
```
pass init GPG-KEY-ID
```
I executed the above with storage name instead of GPG key id. This led to simple string insertion into the `~/.password-store/.gpg-id` instead of key id.
While I think pass should properly check that existing key is used or spit a warning into the output, I still think the mentioned patch is needed.

Thanks!
